scipy/linalg/special_matrices.py

```diff
@@ -1194,3 +1194,66 @@ def fiedler_companion(a):
     c[[0, 1], 0] = [-a[1], 1]
 
     return c
+
+def convmtx(a, n, mode='full'):
+    """
+    Construct a convolution matrix.
+
+    Constructs a matrix
+    `A = convmtx(a, n[, mode])`
+    such that
+    `np.dot(A, v)` is equivalent to `convolve(a, v[, mode])` but slower.
+
+    Parameters
+    ----------
+    a : The array to convolve.
+    n : The number of columns in the resulting matrix.
+        This is analogous to the length of v in numpy.convolve(v, a)
+    mode : This is analogous to `mode` in numpy.convolve(v, a, mode).
+        It determines the number of rows in A is
+            'full'[default]: `len(a) + n - 1`
+            'same': `max(n, len(a) )`
+            'valid': `max(n, len(a) ) - min(n, len(a) ) + 1`
```

Comment:
If you search for the docstring render produced in CI ("build_docs artifact") and navigate to this new function, I think you'll notice that the formatting of these values is not quite right--they get kind of crammed on the same line

Reply:
Ewww! .. Yeah I'll fix that.  

I'll push later tonight or tomorrow addressing your above excellent comments.